pkg/proc/core/minidump/minidump.go

```diff
@@ -0,0 +1,689 @@
+	FileFilterMemory                    FileFlags = 0x00000008
+	FileScanMemory                      FileFlags = 0x00000010
+	FileWithUnloadedModules             FileFlags = 0x00000020
+	FileWithIncorrectlyReferencedMemory FileFlags = 0x00000040
+	FileFilterModulePaths               FileFlags = 0x00000080
+	FileWithProcessThreadData           FileFlags = 0x00000100
+	FileWithPrivateReadWriteMemory      FileFlags = 0x00000200
+	FileWithoutOptionalData             FileFlags = 0x00000400
+	FileWithFullMemoryInfo              FileFlags = 0x00000800
+	FileWithThreadInfo                  FileFlags = 0x00001000
+	FileWithCodeSegs                    FileFlags = 0x00002000
+	FileWithoutAuxilliarySegs           FileFlags = 0x00004000
+	FileWithFullAuxilliaryState         FileFlags = 0x00008000
+	FileWithPrivateCopyMemory           FileFlags = 0x00010000
+	FileIgnoreInaccessibleMemory        FileFlags = 0x00020000
+	FileWithTokenInformation            FileFlags = 0x00040000
+)
+
+// StreamType is the type of the StreamType field of MINIDUMP_DIRECTORY
+type StreamType uint32
+
+const (
+	UnusedStream              StreamType = 0
+	ReservedStream0           StreamType = 1
+	ReservedStream1           StreamType = 2
+	ThreadListStream          StreamType = 3
+	ModuleListStream          StreamType = 4
+	MemoryListStream          StreamType = 5
+	ExceptionStream           StreamType = 6
+	SystemInfoStream          StreamType = 7
+	ThreadExListStream        StreamType = 8
+	Memory64ListStream        StreamType = 9
+	CommentStreamA            StreamType = 10
+	CommentStreamW            StreamType = 11
+	HandleDataStream          StreamType = 12
+	FunctionTableStream       StreamType = 13
+	UnloadedModuleStream      StreamType = 14
+	MiscInfoStream            StreamType = 15
+	MemoryInfoListStream      StreamType = 16
+	ThreadInfoListStream      StreamType = 17
+	HandleOperationListStream StreamType = 18
+	TokenStream               StreamType = 19
+	JavascriptDataStream      StreamType = 20
+	SystemMemoryInfoStream    StreamType = 21
+	ProcessVMCounterStream    StreamType = 22
+)
+
+// Arch is the type of the ProcessorArchitecture field of MINIDUMP_SYSTEM_INFO.
+type Arch uint16
+
+const (
+	CpuArchitectureX86     Arch = 0
+	CpuArchitectureMips    Arch = 1
+	CpuArchitectureAlpha   Arch = 2
+	CpuArchitecturePPC     Arch = 3
+	CpuArchitectureSHX     Arch = 4 // Super-H
+	CpuArchitectureARM     Arch = 5
+	CpuArchitectureIA64    Arch = 6
+	CpuArchitectureAlpha64 Arch = 7
+	CpuArchitectureMSIL    Arch = 8 // Microsoft Intermediate Language
+	CpuArchitectureAMD64   Arch = 9
+	CpuArchitectureWoW64   Arch = 10
+	CpuArchitectureARM64   Arch = 12
+	CpuArchitectureUnknown Arch = 0xffff
+)
+
+// Open reads the minidump file at path and returns it as a Minidump structure.
+func Open(path string, logfn func(fmt string, args ...interface{})) (*Minidump, error) {
+	rawbuf, err := ioutil.ReadFile(path) //TODO(aarzilli): mmap?
+	must(err)
```

Comment:
This convenience function is only used once, let's just inline the error check / panic.